src/uct/api/uct_def.h

```diff
@@ -643,15 +667,19 @@ typedef void (*uct_cm_ep_client_connect_callback_t)(uct_ep_h ep, void *arg,
 
 /**
  * @ingroup UCT_CLIENT_SERVER
- * @brief Callback to handle the disconnection of the remote peer.
+ * @brief Callback to handle the disconnection of the remote peer or a local error.
```

Comment:
is it really possible disconnect will be called for a local error?
can you give example?

Reply:
yes, https://github.com/openucx/ucx/blob/master/src/uct/ib/rdmacm/rdmacm_cm_ep.c#L136
in uct_rdmacm_cm_ep_conn_notify() , if rdma_establish() fails, we call uct_rdmacm_cm_ep_set_failed() which would call the disconnect_cb since the connect flag is set.

Reply:
this seems wrong, since we already return error from uct_rdmacm_cm_ep_conn_notify(), we should not have more than one way to report error.
@evgeny-leksikov wdyt?

Reply:
agree, need to suppress disconnect CB from `uct_rdmacm_cm_ep_conn_notify` since it returns status.